### ISSUE TITLE
Bug: GSF crashes with a segmentation error (single electron, ATLAS production)

### ISSUE THREAD

**Issue by Maury98** [acts-project/acts#1666]
I'm currently doing refitting of ATLAS tracks inside Athena (the ATLAS software environment) with the ACTS GSF, so calling the `fit` function of GSF.
On some events, a crash (segmentation fault) occurs just after calling the `fit` function from [GaussianSumFitter.hpp#L139](https://github.com/acts-project/acts/blob/main/Core/include/Acts/TrackFitting/GaussianSumFitter.hpp#L139) and before this function returns its result, so it does not come from the ATLAS-side code. As of now, I am not able to obtain a verbose output from the GSF fitter ACTS-side.

To reproduce the error, one would need access to ATLAS' files and environment:
1. locally compile https://gitlab.cern.ch/armaury/athena-acts-gsf-refitting/-/tree/gsf_uptodate (branch "gsf_uptodate") with the following packages in the package_filter.txt:
- + Tracking/Acts/ActsTrkAlgs/ActsTrkAnalysis
- + Tracking/Acts/ActsTrkAlgs/ActsTrkFittingAlgs
- + Tracking/Acts/ActsTrkTools/ActsTrkFittingTools
2. Get an RDO (single electron, no pileup) such as from mc15_14TeV.900032.PG_singleel_Pt1_etaFlatnp0_43.recon.RDO.e8185_s3856_r13995 to run on, e.g. the RDO.30737808._000001.pool.root.1 crashes after having processed 163 events, so at event n°164.
3. Run the following Reco_tf.py:
```
Reco_tf.py --CA --autoConfiguration 'everything' --steering doRAWtoALL --detectors ITkStrip ITkPixel Bpipe --preExec "ConfigFlags.Reco.EnableHGTDExtension = False;" \
  --preInclude "InDetConfig.ConfigurationHelpers.OnlyTrackingPreInclude" --postInclude "ActsTrkFittingAlgs.ActsTrkFittingAlgsConfig.ActsReFitterGSFAlgCfg" \
  --inputRDOFile ... --outputAODFile ... --maxEvents -1
```

[Bug]

**Comment by benjaminhuth** [comment]
So without a stack trace or verbose log it is difficult to say where the error exactely comes from. That would be the first step. What is the exact issue with the VERBOSE output?

It might be possible that it will be fixed by the large GSF refactor I'm currently doing (#1629). Is it easy possible to compile ACTS with this branch and check if it is still present

**Comment by Maury98** [comment]
I run the Reco_tf.py with those additionnal flags: `--verbose --multithreaded=False --multiprocess=False`.
Here is a link to the log file when running the above command (`Reco_tf.py ...`) with the above mentionned RDO file: https://cernbox.cern.ch/s/ikiPkpvJVFSULGt

**Comment by Maury98** [comment]
Yes, I'm currently looking into compiling and using a custom ACTS build with Athena. I'll try using your refactor/gsf-big-refactoring branch from your fork then

**Comment by benjaminhuth** [comment]
Ah here we go:

RAWtoALL 10:44:30 ActsReFitterAlg.ActsKalma...Acts GaussianSum RefitWARNING Did not find forward states on surface vol=21|lay=4|sen=2676

This should be fixed with #1629
